subsys/bluetooth/controller/ll_sw/nordic/lll/lll_central_iso.c

```diff
@@ -713,7 +674,25 @@ static void isr_rx(void *param)
 	cis_lll = param;
 
 	/* No Rx */
-	if (!trx_done) {
+	if (!trx_done ||
+#if defined(CONFIG_TEST_FT_CEN_SKIP_SUBEVENTS)
+	    /* Used by test code,
+	     * to skip a number of events in every 3 event count when current subevent is less than
+	     * or equal to 2 or when current subevent has completed all its NSE number of subevents.
+	     * OR
+	     * to skip a (number + 1) of events in every 3 event count when current subevent is less
+	     * than or equal to 1 or when current subevent has completed all its NSE number of
+	     * subevents.
+	     */
+	    ((((cis_lll->event_count % 3U) < CONFIG_TEST_FT_CEN_SKIP_EVENTS_COUNT) &&
+	      ((se_curr > cis_lll->nse) || (se_curr <= 2U))) ||
+
+	     (((cis_lll->event_count % 3U) < (CONFIG_TEST_FT_CEN_SKIP_EVENTS_COUNT + 1U)) &&
+	      ((se_curr > cis_lll->nse) || (se_curr <= 1U)))) ||
+#endif
```

Comment:
I don't understand why we have this block of embedded test code. Can you elaborate?

Reply:
I'm sorry but I still do not understand _why_ we have this. 

You seem to modify the behavior of the stack depending on whether some testing is enabled, which seems odd to me.

Reply:
This Kconfig is only defined in the bsim test project, and is not available to users writing samples/applications.

Reply:
So we have code in the subsystem that is only enabled by a specific application? :s 

> is not available to users writing samples/applications.

Except that it is if they simply add that Kconfig themselves. 

Will need input from others to see if this is a good way of doing it, because to me it doesn't seem like a good idea (generally we don't have test-specific code in the subsystem).

Reply:
@kruithofa @wopu-ot @thoh-ot comments welcome on how to handle invalid behavior testing in babblesim?